src/vellum/workflows/graph/graph.py

```diff
@@ -53,9 +55,13 @@ def from_set(targets: GraphTargetOfSets) -> "Graph":
                 entrypoints.update(target._entrypoints)
                 edges.update(target._edges)
                 terminals.update(target._terminals)
-            else:
+            elif hasattr(target, "Ports"):
```

Comment:
we can't use `isinstance(Port)` or `issubclass(BaseNode)` bc they would introduce circular imports so they remain in dynamic TYPE_CHECKING imports above